connector/oidc/oidc.go

```diff
@@ -215,10 +225,57 @@ func (c *oidcConnector) HandleCallback(s connector.Scopes, r *http.Request) (ide
 		Email:         claims.Email,
 		EmailVerified: claims.EmailVerified,
 	}
+
+	// Add AccessToken to user identity for future requests
+	connData, err := json.Marshal(connectorData{
+		AccessToken: token.AccessToken,
+	})
+	if err != nil {
+		return identity, fmt.Errorf("marshal connector data: %v", err)
+	}
+
+	identity.ConnectorData = connData
+
 	return identity, nil
 }
 
 // Refresh is implemented for backwards compatibility, even though it's a no-op.
 func (c *oidcConnector) Refresh(ctx context.Context, s connector.Scopes, identity connector.Identity) (connector.Identity, error) {
 	return identity, nil
 }
+
+func (c *oidcConnector) GetUserInfo(connData []byte, user *map[string]interface{}) (err error) {
+
+	// Extract Access Token that was stored while the connector handled the OIDC handshake
+	var cData connectorData
+
+	if err := json.Unmarshal(connData, &cData); err != nil {
+		c.logger.Errorf("Failed to read connector data : %v", err)
+		return err
+	}
+
+	// Format Authorization header to request user info on behalf of the client
+	authorization := fmt.Sprintf("bearer %s", cData.AccessToken)
+	// Prepare get request including Authorization header from RP
+	req, err := http.NewRequest("GET", c.userInfoURI, nil)
+
```

Comment:
Nit: This blank line is kinda odd, would expect the `if err != nil` to directly follow where the error is created